driver-core/src/test/unit/com/mongodb/internal/connection/ScramShaAuthenticatorSpecification.groovy

```diff
@@ -442,6 +494,17 @@ class ScramShaAuthenticatorSpecification extends Specification {
         validateClientMessages(connection, clientMessages, authenticator.getMechanismName())
     }
 
+    def validateSpeculativeAuthentication(String payloads, ScramShaAuthenticator authenticator, boolean async,
+                                          BsonDocument speculativeAuthenticateResponse) {
+        def (clientMessages, serverResponses) = createMessages(payloads, false)
+        def connection = createConnection(serverResponses, 0)
+        authenticator.createSpeculativeAuthenticateCommand(connection)
```

Comment:
Let's assert somewhere (it has to be back up in a Spock-ish method) that this returns the expected document